File: python/Leetcode374.py
class Leetcode374:
    def guessNumber(self, n: int) -> int:
        l = 1
        r = n
        # arbitrary function
        def guess(num):
            return 0
        while l <= r:
            mid = (l + r) // 2
            res = guess(mid)            
            if res == -1:
                r = mid - 1
            elif res == 1:
                l = mid + 1
            else:
                return mid
        return 0
        

File: java/src/Leetcode374.java
public class Leetcode374 {
    public int guessNumber(int n) {
        int l = 1, r = n;
        while(l <= r){
            int mid = (l - r) / 2 + r;
            if(guess(mid) == -1){
                r = mid - 1;
            }else if(guess(mid) == 1){
                l = mid + 1;
            }else{
                return mid;
            }
        }
        return 0;
    }
}
